### ISSUE TITLE
Docs on building firmware

### ISSUE THREAD

**Issue by ian-fox** [hak5darren/USB-Rubber-Ducky#171]
Hello,

I'm trying to build the firmware to do some hacking on it, but having some trouble. I see there's a wiki page on flashing the ducky which looks like it will be helpful once I've built the firmware, but doesn't cover the actual compilation.

So far I've noticed that there are some makefiles which say they're autogenerated (but not what they're autogenerated from), though they don't seem to work, and some project files which look like they're from Atmel Studio, but when opened there it spits out 671 errors which I'm not super keen on debugging right now without the help of somebody who's done this before.

Wondering if anybody who's gone through this can point me in the right direction.

Thanks.

**Comment by makozort** [comment]
Not sure if this is what your looking for but look at this https://ducktoolkit.com/encode

**Comment by ian-fox** [comment]
Not quite, I'm interested in building the code in the `firmware` directory of this repo, which is the thing that takes `inject.bin` files (e.g. from the ducktoolkit encoder) and runs them. Thanks though!
